lib/rules/no-extra-parens.js

```diff
@@ -169,6 +169,26 @@ module.exports = {
             return ruleApplies(node) && isParenthesisedTwice(node);
         }
 
+        /**
+         * Determines if a node that is expected to be parenthesised is surrounded by
+         * (potentially) invalid extra parentheses with considering precedence.
+         * @param {ASTNode} node The node to be checked.
+         * @param {number} precedenceLowerLimit The lower limit of precedence.
+         * @returns {boolean} True if the node is has an unexpected extra pair of parentheses.
+         * @private
+         */
+        function hasExcessParensWithConsiderPrecedence(node, precedenceLowerLimit) {
```

Comment:
I wonder if we could rename this to `hasExcessParensWithPrecedence` or ``hasExcessParensWhileConsideringPrecedence` (I prefer the first)?